src/app/modules/item/pages/item-display/item-display.component.ts

```diff
@@ -140,26 +147,32 @@ export class ItemDisplayComponent implements OnInit, AfterViewInit, OnDestroy {
 }
 
 export class ItemDisplayPlatform extends Platform {
```

Comment:
I don't have the full picture yet but why not having Platform (or TaskListener) to be an interface and having `ItemDisplayComponent` implement this interface directly? Intuitively I think that's how I would have done it but maybe there are reasons to do it this way.

Reply:
I wasn't sure whether it's clean to have a component also implement another interface which is unrelated to it being a component. I'll do that.

Reply:
I seems much cleaner if I haven't missed an aspect.  And so the "Platform" would just be an interface. 
In this context, I think the interface should not be called "Platform" but more "TaskListener"... IMO that's not a problem if the api method mention "platform" for that, which is the correct term in the API, but in the platform itself having a "platform" concept to represent oneself is quite confusing.

Reply:
Renamed to TaskListener. Note that as discussed, the TaskListener functions in `task-proxy.ts` will be merged into the item-display component.